lib/tapioca/dsl/compilers/active_record_relations.rb

```diff
@@ -991,25 +991,50 @@ def create_common_methods
           end
 
           FIND_OR_CREATE_METHODS.each do |method_name|
-            block_type = "T.nilable(T.proc.params(object: #{constant_name}).void)"
-            create_common_method(
-              method_name,
-              parameters: [
-                create_param("attributes", type: "T.untyped"),
-                create_block_param("block", type: block_type),
-              ],
-              return_type: constant_name,
+            sigs = [
+              common_relation_methods_module.create_sig(
```

Comment:
Do you mind leaving a comment on top of this sig (or the `T.untyped` sig) explaining why it needs to be the first entry (or the last entry, if you want to do it for the other one)?

It will be helpful for future onlookers to know that there is intent behind the order of the sigs.